src/main/resources/jenkins/security/plugins/ldap/Messages.properties

```diff
@@ -1,6 +1,7 @@
 FromUserRecordLDAPGroupMembershipStrategy.DisplayName=Parse user attribute for list of LDAP groups
 FromGroupSearchLDAPGroupMembershipStrategy.DisplayName=Search for LDAP groups containing user
 LDAPSecurityRealm.AtLeastOne=At least one LDAP server configuration must be configured.
+LDAPSecurityRealm.NotSameServer=Each server configuration must have a unique server string
```

Comment:
So with this solution I cannot have two different configurations in the same LDAP server without resorting to port trickery or a DNS name alias.

This would seem to be in conflict with the use case in the original request:

> After a company merger/consolidation there is a single authentication system but the schemas have not been merged.

In that case, there would be a single LDAP "server" but it might have two distinct trees with two distinct manager passwords.

I would think that the correct solution is to use a composite key comprising the server, root DN, user search base and user filter. It may be necessary to tweak how that gets consolidated e.g. to handle the case where the combined root DN + user search base is the same effective user search base as another.

However we would need to check with @cyrille-leclerc as to whether there is a need to specify multiple configurations for the same LDAP server. The agreed acceptance criteria do not list that as necessary, but Cyrille was not part of a min-max of the AC